src/tfs/tlog.c

```diff
@@ -349,8 +541,12 @@ closure_function(1, 1, void, log_flush_timer_expired,
 
 static void log_set_dirty(log tl)
 {
-    if (tl->dirty)
+    if (tl->dirty) {
+        if (buffer_length(tl->tuple_staging) >= bytes_from_sectors(tl->fs,
+                range_span(tl->current->sectors)) / 2)
+            log_flush(tl, 0);
```

Comment:
can cancel timer here

Reply:
The timer is already being canceled in log_flush()